x/dispensation/handler.go

```diff
@@ -12,8 +12,8 @@ func NewHandler(k Keeper) sdk.Handler {
 	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
 		ctx = ctx.WithEventManager(sdk.NewEventManager())
 		switch msg := msg.(type) {
-		case MsgDistribution:
-			return handleMsgCreateDistribution(ctx, k, msg)
+		case *MsgDistribution:
+			return handleMsgCreateDistribution(ctx, k, *msg)
```

Comment:
you may want to use the message server here. Not urgent, it should just be known that all txs are amino handled for this module.